src/Quasar.jl

```diff
@@ -1427,10 +1434,13 @@ function evaluate_qubits(v::AbstractVisitor, qubit_targets)::Vector{Int}
     qubits = Iterators.flatmap(qubit_targets) do qubit_expr
         if head(qubit_expr) == :identifier
             qubit_name = name(qubit_expr)
+            haskey(mapping, qubit_name) || throw(QasmVisitorError("Missing input variable '$qubit_name'.", "NameError"))
             return mapping[qubit_name]
         elseif head(qubit_expr) == :indexed_identifier
             qubit_name = name(qubit_expr)
             qubit_ix   = evaluate(v, qubit_expr.args[2])
+            # Need to figure this out
```

Comment:
The issue here might be that `qubit_ix` can itself be an array? You might need to index over it too